pytorch_lightning/loops/dataloader/evaluation_dataloader_loop.py

```diff
@@ -0,0 +1,268 @@
+                self.trainer.num_sanity_val_batches = [
+                    min(self.trainer.num_sanity_val_steps, val_batches) for val_batches in self.trainer.num_val_batches
+                ]
+                max_batches = self.trainer.num_sanity_val_batches
+            else:
+                max_batches = self.trainer.num_val_batches
+        return max_batches
+
+    def reload_evaluation_dataloaders(self) -> None:
+        """Reloads dataloaders if necessary"""
+        model = self.trainer.lightning_module
+        if self.trainer.testing:
+            self.trainer.reset_test_dataloader(model)
+        elif self.trainer.val_dataloaders is None or self.trainer.reload_dataloaders_every_epoch:
+            self.trainer.reset_val_dataloader(model)
+
+    def on_evaluation_start(self, *args: Any, **kwargs: Any) -> None:
+        """Runs ``on_{validation/test}_start`` hooks"""
+        self.should_track_batch_outputs_for_epoch_end: bool = self._should_track_batch_outputs_for_epoch_end()
+
+        assert self.results is not None
+        self.results.to(device=self.trainer.lightning_module.device)
+
+        if self.trainer.testing:
+            self.trainer.call_hook("on_test_start", *args, **kwargs)
+        else:
+            self.trainer.call_hook("on_validation_start", *args, **kwargs)
+
+    def on_evaluation_model_eval(self) -> None:
+        """Sets model to eval mode"""
+        model_ref = self.trainer.lightning_module
+        if self.trainer.testing:
+            model_ref.on_test_model_eval()
+        else:
+            model_ref.on_validation_model_eval()
+
+    def on_evaluation_model_train(self) -> None:
+        """Sets model to train mode"""
+        model_ref = self.trainer.lightning_module
+        if self.trainer.testing:
+            model_ref.on_test_model_train()
+        else:
+            model_ref.on_validation_model_train()
+
+    def on_evaluation_end(self, *args: Any, **kwargs: Any) -> None:
+        """Runs ``on_{validation/test}_end`` hook"""
+        if self.trainer.testing:
+            self.trainer.call_hook("on_test_end", *args, **kwargs)
+        else:
+            self.trainer.call_hook("on_validation_end", *args, **kwargs)
+
+        if self.trainer.state.fn != TrainerFn.FITTING:
+            # summarize profile results
+            self.trainer.profiler.describe()
+
+        # reset any `torchmetrics.Metric` and the logger connector state
+        self.trainer.logger_connector.reset(metrics=True)
+
+    def on_evaluation_epoch_start(self, *args: Any, **kwargs: Any) -> None:
+        """Runs ``on_epoch_start`` and ``on_{validation/test}_epoch_start`` hooks"""
+        self.trainer.logger_connector.on_epoch_start()
+        self.trainer.call_hook("on_epoch_start", *args, **kwargs)
+
+        if self.trainer.testing:
+            self.trainer.call_hook("on_test_epoch_start", *args, **kwargs)
+        else:
+            self.trainer.call_hook("on_validation_epoch_start", *args, **kwargs)
```

Comment:
```suggestion
        hook_name = "on_test_epoch_start" if self.trainer.testing else "on_validation_epoch_start"
        self.trainer.call_hook(hook_name, *args, **kwargs)
```

Reply:
this stuff came from the old loop but ok, we can change it here